src/Processor/Braintree/Mapper/Subscription/ModifierMapper.php

```diff
@@ -112,8 +117,17 @@ private function toModifiersRequest(Modifiers $modifiers, Plan $plan): array
             $modifier = $operation->getModifier();
             $type = $operation->getType();
             $default = $plan->getModifier($modifier->getId());
+            $isAdding = null === $default;
+
+            // when changing plans, allow modifications to default AddOns
+            if (!$subscription->isNew()) {
+                $defaultAddOns = array_keys($subscription->getPlan()->getAddOns());
```

Comment:
Anything with the `HasModifiers` trait can call `getAddOn` or `getDiscount` with an id. The `array_keys`/`in_array` approach seems unnecessary. Essentially:

```php
$defaultAddOns = $subscription->getPlan()->getAddOns();
$isDefaultAddOn = isset($defaultAddOns[$modifier->getId()]);
```

It also looks like this needs to account for discounts.